File: web/console/src/user/service.ts
// Copyright (C) 2021 Creditor Corp. Group.
// See LICENSE for copying information.

import { UserClient } from '@/api/user';
import { User } from '.';

/** exposes all user related logic  */
export class UserService {
    private readonly users: UserClient;
    /** user service depend on user client  */
    public constructor(users: UserClient) {
        this.users = users;
    };
    /** handles user registration */
    public async register(user: User): Promise<void> {
        await this.users.register(user);
    };
    /** return registred user */
    public async login(email: string, password: string): Promise<void> {
        await this.users.login(email, password);
    };
    /** handles user changing password */
    public async changePassword(password: string, newPassword: string): Promise<void> {
        await this.users.changePassword(password, newPassword);
    };
    /** handles user check email token */
    public async checkEmailToken(token: string | null): Promise<void> {
        await this.users.checkEmailToken(token);
    };
    /** handles user check recover token */
    public async checkRecoverToken(token: string | null): Promise<void> {
        await this.users.checkRecoverToken(token);
    };
    /** handles user recover password */
    public async recoverPassword(password: string): Promise<void> {
        await this.users.recoverPassword(password);
    };
};

File: pkg/velas/service.go
// Copyright (C) 2021 Creditor Corp. Group.
// See LICENSE for copying information.

package velas

// Service is handling velas related logic.
//
// architecture: Service
type Service struct {
	config Config
}

// NewService is a constructor for velas service.
func NewService(config Config) *Service {
	return &Service{
		config: config,
	}
}

// Get returns va client fields.
func (service *Service) Get() VAClientFields {
	return VAClientFields{
		ClientID:                   service.config.ClientID,
		RedirectURI:                service.config.RedirectURI,
		AccountProviderHost:        service.config.AccountProviderHost,
		NetworkAPIHost:             service.config.NetworkAPIHost,
		TransactionsSponsorAPIHost: service.config.TransactionsSponsorAPIHost,
		TransactionsSponsorPubKey:  service.config.TransactionsSponsorPubKey,
	}
}
